pkg/cstorpool/v1alpha2/cstorpool.go

```diff
@@ -69,30 +90,81 @@ func (l *cspList) FilterUIDs(p ...predicate) []string {
 // listBuilder enables building a
 // list of csp instances
 type listBuilder struct {
-	list *cspList
+	list *CSPList
 }
 
 // ListBuilder returns a new instance of
 // listBuilder object
 func ListBuilder() *listBuilder {
-	return &listBuilder{list: &cspList{}}
+	return &listBuilder{list: &CSPList{}}
 }
 
 // WithUIDs builds a list of cstor pools
 // based on the provided pool UIDs
 func (b *listBuilder) WithUIDs(poolUIDs ...string) *listBuilder {
 	for _, uid := range poolUIDs {
-		obj := &apis.CStorPool{}
-		obj.SetUID(types.UID(uid))
-		item := &csp{object: obj}
-		b.list.items = append(b.list.items, item)
+		obj := &csp{&apis.CStorPool{}}
+		obj.object.SetUID(types.UID(uid))
+		b.list.items = append(b.list.items, obj)
+	}
+	return b
+}
+
+// WithUIDNodeMap builds a cspList based on the provided
+// map of uid and nodename
+func (b *listBuilder) WithUIDNodeMaps(UIDNodeMap map[string]string) *listBuilder {
+	for k, v := range UIDNodeMap {
+		obj := &csp{&apis.CStorPool{}}
+		obj.object.SetUID(types.UID(k))
+		obj.object.SetAnnotations(map[string]string{string(scheduleOnHostAnnotation): v})
+		b.list.items = append(b.list.items, obj)
 	}
 	return b
 }
 
+// WithCStorPool builds the list based on the provided
+// *apis.CStorPool instances
+func (b *listBuilder) WithList(pools *CSPList) *listBuilder {
+	if pools == nil {
+		return b
+	}
+	b.list.items = append(b.list.items, pools.items...)
+	return b
+}
+
 // List returns the list of csp
 // instances that were built by
 // this builder
-func (b *listBuilder) List() *cspList {
+func (b *listBuilder) List() *CSPList {
 	return b.list
 }
+
+// GetPoolUIDs retuns the UIDs of the pools
+// available in the list
+func (l *CSPList) GetPoolUIDs() []string {
+	uids := []string{}
+	for _, pool := range l.items {
+		uids = append(uids, string(pool.object.GetUID()))
+	}
+	return uids
+}
+
+// newListFromUIDNode exposes WithUIDNodeMap
+// to CAS Templates
+func newListFromUIDNode(UIDNodeMap map[string]string) *CSPList {
```

Comment:
Please remove `map` from name of the argument

Reply:
removed